### PR TITLE
Feature/new app header

### PR DESCRIPTION
**Pull request by jegasriskantha**
relates to https://github.com/IPG-Automotive-UK/cloud-licensing-web/issues/274

## Changes

- Add new generic app header that will be used in the cloud licensing app (and a generic layout coming soon) . This app header can also be used in the VIRTO homepage. 

- The generic app header builds on the the VIRTO app header but has the IPG logo in the top left, the app name in the center and does not have the dark mode toggle embedded. 
- In order to make this generic app header more universal a new children property allows users to add child components that will render on the right side of the header but to left of the user menu / avatar. 
- The appName is now optional and there is a new property for appLogo. If both an appName and a appLogo are provided the appLogo takes priority.

## Do Not Merge
- This PR builds of work done here: https://github.com/IPG-Automotive-UK/react-ui/pull/585 and should not be merged till this is merged. 

## UI/UX
- This is the new header with default props
image

- This is an example of the header configured to look similiar to the VIRTO homepage:
image

- This is an example of the header configured to look similar to the cloud licensing header
image

Here is a link to the cloud licensing Figma Prototype:  https://www.figma.com/proto/O3IAobLTPcjIO1WaxaPod6/IPG-Cloud-Licensing?node-id=68-14265&starting-point-node-id=68%3A14265

## Author checklist before assigning a reviewer

- [x] Reviewed my own code-diff.
- [ ] Branch has been run in docker.
- [x] Assigned to me.
- [x] Relevant tags added.
- [x] Appropriate tests have been added.
- [x] Lint and test workflows pass.

### CODE REVIEW

File: src/AppHeader/AppHeader.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      <SearchBar />
+      <ToggleColorMode mode={"light"} onChange={() => {}} />
+    </>
+  ),
+  mode: "light",
+  username: "Ruud van Nistelrooy"
+};
+
+/**
+ * Story for the AppHeader component where a multiple children are passed
```
Comment (m4manjesh) [inline src/AppHeader/AppHeader.stories.tsx:82]: This story includes a logo as well. so I think you can update the comment accordingly.

**Review by m4manjesh — CHANGES_REQUESTED** [review]

File: src/AppHeader/AppHeader.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+/**
+ * Story for the AppHeader component where a single child is passed
+ */
+export const SingleChild = Template.bind({});
+SingleChild.args = {
+  appName: "APP NAME",
+  children: (
+    <IconButton sx={{ mr: 1 }}>
+      <Notifications color="info" fontSize="large" />
```
[the remark this answers is not in the document]
Comment (jegasriskantha) [reply src/AppHeader/AppHeader.stories.tsx:58]: This is just an example in the story, the user will be able to set any component as a child to the app header, that's why it doesn't have functionality.

File: src/AppHeader/AppHeader.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+/**
+ * Story for the AppHeader component where a multiple children are passed
+ */
+export const MultipleChildren = Template.bind({});
+MultipleChildren.args = {
+  appName: "APP NAME",
+  children: (
+    <>
+      <SearchBar />
+      <ToggleColorMode mode={"light"} onChange={() => {}} />
```
[the remark this answers is not in the document]
Comment (jegasriskantha) [reply src/AppHeader/AppHeader.stories.tsx:74]: This is just an example in the story, the user will be able to set any component as a child to the app header, that's why it doesn't have functionality.

File: src/AppHeader/AppHeader.stories.tsx (hunk 1 above)
Comment (jegasriskantha) [reply src/AppHeader/AppHeader.stories.tsx]: resolved here: https://github.com/IPG-Automotive-UK/react-ui/pull/602/commits/fa335b95a86488adc1498d90cd1268bc64a0f344

**Review by m4manjesh — APPROVED** [review]
LGTM. Please take care of merge and release

**Pull request merged**